### PR TITLE
feat(ui): login via email and password - Basic auth

### PR DESCRIPTION
**Pull request by chirag-madlani**
### Describe your changes :

I worked on the supporting #6894

#
### Type of change :

- [x] Improvement

#
### Frontend Preview (Screenshots) :
1. Login

Screenshot 2022-09-19 at 7 51 06 PM

2. Forgot password
Screenshot 2022-09-19 at 7 51 26 PM

3. Signup
Screenshot 2022-09-19 at 7 51 49 PM

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx [hunk 1]
```diff
@@ -82,7 +81,7 @@ interface AuthProviderProps {
 
 const cookieStorage = new CookieStorage();
 
-const userAPIQueryFields = 'profile,teams,roles';
+const userAPIQueryFields = 'profile,teams,roles,isEmailVerified';
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx:84]: Can we move this to constant?

File: openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx [hunk 2]
```diff
@@ -392,7 +395,7 @@ export const AuthProvider = ({
   const initializeAxiosInterceptors = () => {
     // Axios Request interceptor to add Bearer tokens in Header
     axiosClient.interceptors.request.use(async function (config) {
-      const token: string | void = localStorage.getItem(oidcTokenKey) || '';
+      const token: string | void = localState.getOidcToken() || '';
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx:398]:
```suggestion
      const token: string = localState.getOidcToken() || '';
```

File: openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/basic-auth.provider.tsx [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+import localState from '../../utils/LocalStorageUtils';
+import {
+  showErrorToast,
+  showInfoToast,
+  showSuccessToast,
+} from '../../utils/ToastUtils';
+import { useAuthContext } from './AuthProvider';
+import { OidcUser } from './AuthProvider.interface';
+
+export interface JWTPayloadV1 extends JwtPayload {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/basic-auth.provider.tsx:26]:
```suggestion
export interface BasicAuthJWTPayload extends JwtPayload {
```

File: openmetadata-ui/src/main/resources/ui/src/components/common/auth-common-card/auth-common-card.component.tsx [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import { Card } from 'antd';
+import React from 'react';
+import SVGIcons, { Icons } from '../../../utils/SvgUtils';
+
+const AuthCommonCard = ({
+  children,
+  classNames,
+}: {
+  children: JSX.Element;
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/common/auth-common-card/auth-common-card.component.tsx:9]:
```suggestion
  children: ReactNode;
```

File: openmetadata-ui/src/main/resources/ui/src/constants/auth.constants.ts [hunk 5]
```diff
@@ -12,4 +12,20 @@
  */
 
 export const NO_AUTH = 'no-auth';
-export const NOOP_FILTER = 'org.openmetadata.service.security.NoopFilter';
+export const NOOP_FILTER = 'org.openmetadata.catalog.security.NoopFilter';
+
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/constants/auth.constants.ts:16]:
```suggestion
export const NOOP_FILTER = 'org.openmetadata.service.security.NoopFilter';

```

File: openmetadata-ui/src/main/resources/ui/src/pages/forgot-password/forgot-password.component.tsx [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+const ForgotPassword = () => {
+  const { isPasswordResetLinkSent, handleForgotPassword } = useBasicAuth();
+  const [hasError, setHasError] = useState(false);
+  const handleSubmit = (data: { email: string }) => {
+    handleForgotPassword && handleForgotPassword(data.email);
+  };
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  const onError = (errors: any) => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/forgot-password/forgot-password.component.tsx:16]: Can we use the proper type here?

File: openmetadata-ui/src/main/resources/ui/src/pages/forgot-password/forgot-password.component.tsx [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+    setHasError(errors.email);
+  };
+
+  return (
+    <div className="h-full tw-py-36">
+      <Card
+        bodyStyle={{ padding: '48px' }}
+        className=" m-y-lg m-auto p-x-lg"
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/forgot-password/forgot-password.component.tsx:24]:
```suggestion
        className="m-y-lg m-auto p-x-lg"
```

File: openmetadata-ui/src/main/resources/ui/src/pages/forgot-password/forgot-password.styles.less [hunk 8]
```diff
@@ -0,0 +1,14 @@
...
+input.forgot-email-input {
+  border-color: #ff4c3b;
+}
+input.forgot-email-input:focus {
+  outline: none;
+  border-color: 1px solid #ff4c3b;
+}
+input.forgot-email-input:hover {
+  border-color: #ff4c3b;
+}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/forgot-password/forgot-password.styles.less:14]:
```suggestion
input.forgot-email-input {
  border-color: #ff4c3b;
  &:focus {
   outline: none;
   border-color: 1px solid #ff4c3b;
  };
  &:hover{
   border-color: #ff4c3b;
  }
}
```

File: openmetadata-ui/src/main/resources/ui/src/pages/login/index.tsx [hunk 9]
```diff
@@ -136,19 +151,84 @@ const SigninPage = () => {
...
 
+  const handleSubmit = async (data: { email: string; password: string }) => {
+    handleLogin(data.email, data.password);
+  };
+
+  const onClickSignUp = () => history.push(ROUTES.REGISTER);
+
+  const onClickForgotPassword = () => history.push(ROUTES.FORGOT_PASSWORD);
+
+  const validationMessages = {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/login/index.tsx:162]: Can we move this to constants?

File: openmetadata-ui/src/main/resources/ui/src/pages/login/index.tsx [hunk 10]
```diff
@@ -136,19 +151,84 @@ const SigninPage = () => {
...
+                  <Button type="primary" onClick={onClickSignUp}>
+                    Sign Up
+                  </Button>
+                </div>
+              </div>
+            ) : null}
+
+            {!isAuthProviderBasic ? (
+              <div className="">{getSignInButton()}</div>
+            ) : null}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/login/index.tsx:231]:
```suggestion
            ) : <div className="">{getSignInButton()}</div>}
```

File: openmetadata-ui/src/main/resources/ui/src/pages/login/login.style.less [hunk 11]
```diff
@@ -0,0 +1,11 @@
+.ant-form-item-explain-error {
+  text-align: left;
+}
+
+.ant-divider-horizontal .ant-divider-with-text::before,
+.ant-divider-horizontal .ant-divider-with-text::after {
+  width: 20%;
+}
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/login/login.style.less:8]:
```suggestion
.ant-divider-horizontal .ant-divider-with-text {
&:before, &:after{
  width: 20%;
  }
}
```

File: openmetadata-ui/src/main/resources/ui/src/pages/reset-password/reset-password.component.tsx [hunk 12]
```diff
@@ -0,0 +1,135 @@
...
+      token: params?.token,
+      username: params?.userName,
+      password: data.password,
+      confirmPassword: data.confirmPassword,
+    } as PasswordResetRequest;
+
+    handleResetPassword(ResetRequest);
+  };
+
+  const validationMessages = {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/pages/reset-password/reset-password.component.tsx:44]: Can we move this to conatnts?

**Review by ShaileshParmar11 — APPROVED** [review]

**Pull request merged**